pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -86,19 +119,68 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		return err
 	}
 
-	mapper, typer := f.Object()
-	// TODO: use resource.Builder instead
-	obj, err := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
-		NamespaceParam(cmdNamespace).RequireNamespace().
-		FilenameParam(filename).
-		Do().
-		Object()
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+
+	// fetch rc
+	oldRc, err := client.ReplicationControllers(cmdNamespace).Get(oldName)
 	if err != nil {
 		return err
 	}
-	newRc, ok := obj.(*api.ReplicationController)
-	if !ok {
-		return cmdutil.UsageError(cmd, "%s does not specify a valid ReplicationController", filename)
+
+	mapper, typer := f.Object()
+	var newRc *api.ReplicationController
+
+	if len(filename) != 0 {
+		obj, err := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+			NamespaceParam(cmdNamespace).RequireNamespace().
+			FilenameParam(filename).
+			Do().
+			Object()
+		if err != nil {
+			return err
+		}
+		var ok bool
+		newRc, ok = obj.(*api.ReplicationController)
+		if !ok {
+			return cmdutil.UsageError(cmd, "%s does not specify a valid ReplicationController", filename)
+		}
+	}
+	if len(image) != 0 {
+		newName := args[1]
+		var err error
+		// load the old RC into the "new" RC
+		if newRc, err = client.ReplicationControllers(cmdNamespace).Get(oldName); err != nil {
+			return err
+		}
+
+		if len(newRc.Spec.Template.Spec.Containers) > 1 {
+			// TODO: support multi-container image update.
+			return fmt.Errorf("Image update is not supported for multi-container pods")
+		}
+		if len(newRc.Spec.Template.Spec.Containers) == 0 {
+			return cmdutil.UsageError(cmd, "Pod has no containers! (%v)", newRc)
+		}
+		newRc.Spec.Template.Spec.Containers[0].Image = image
+		newRc.Spec.Replicas = 0
```

Comment:
Replicas will just be set back to oldRc.Spec.Replicas, below, so this seems pointless.